pkg/controller/packages/pkg/pkg_test.go

```diff
@@ -1412,7 +1421,88 @@ func TestProcessDeployment(t *testing.T) {
-						withDeploymentSecurityContext(&corev1.PodSecurityContext{RunAsNonRoot: &runAsNonRoot}),
+						withDeploymentSecurityContext(&corev1.PodSecurityContext{
+							RunAsNonRoot: &runAsNonRoot,
+						}),
+						withDeploymentContainerSecurityContext(&corev1.SecurityContext{
+							AllowPrivilegeEscalation: &allowPrivilegeEscalation,
+							Privileged:               &privileged,
+							RunAsNonRoot:             &runAsNonRoot,
+						}),
+					),
+				},
+				controllerRef: &corev1.ObjectReference{
+					Name:       controllerDeploymentName,
+					Namespace:  namespace,
+					Kind:       "Deployment",
+					APIVersion: "apps/v1",
+				},
+			},
+		},
+		{
+			name: "SuccessNonZeroFsGroup",
+			r: resource(withControllerSpec(defaultControllerSpec(
+				withDeploymentSecurityContext(&corev1.PodSecurityContext{
+					FSGroup: &nobodyGroup,
+				}),
+			))),
+			clientFunc: fake.NewFakeClient,
+			want: want{
+				err: nil,
+				d: &apps.Deployment{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      controllerDeploymentName,
+						Namespace: namespace,
+						Labels:    packagespkg.ParentLabels(resource(withControllerSpec(defaultControllerSpec()))),
+					},
+					Spec: *deploymentSpec(
+						withDeploymentTmplMeta(controllerDeploymentName, "", nil),
+						withDeploymentMatchLabels(map[string]string{"app": controllerDeploymentName}),
+						withDeploymentSA(resourceName),
+						withDeploymentContainer(controllerContainerName, controllerImageName),
+						withDeploymentSecurityContext(&corev1.PodSecurityContext{
+							RunAsNonRoot: &runAsNonRoot,
+							FSGroup:      &nobodyGroup,
+						}),
+						withDeploymentContainerSecurityContext(&corev1.SecurityContext{
+							AllowPrivilegeEscalation: &allowPrivilegeEscalation,
+							Privileged:               &privileged,
+							RunAsNonRoot:             &runAsNonRoot,
+						}),
+					),
+				},
+				controllerRef: &corev1.ObjectReference{
+					Name:       controllerDeploymentName,
+					Namespace:  namespace,
+					Kind:       "Deployment",
+					APIVersion: "apps/v1",
+				},
+			},
+		},
+		{
+			name: "SuccessZeroFsGroup",
+			r: resource(withControllerSpec(defaultControllerSpec(
+				withDeploymentSecurityContext(&corev1.PodSecurityContext{
+					FSGroup: &rootGroup,
+				}),
+			))),
+			clientFunc: fake.NewFakeClient,
+			want: want{
+				err: nil,
+				d: &apps.Deployment{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      controllerDeploymentName,
+						Namespace: namespace,
+						Labels:    packagespkg.ParentLabels(resource(withControllerSpec(defaultControllerSpec()))),
+					},
+					Spec: *deploymentSpec(
+						withDeploymentTmplMeta(controllerDeploymentName, "", nil),
+						withDeploymentMatchLabels(map[string]string{"app": controllerDeploymentName}),
+						withDeploymentSA(resourceName),
+						withDeploymentContainer(controllerContainerName, controllerImageName),
+						withDeploymentSecurityContext(&corev1.PodSecurityContext{
+							RunAsNonRoot: &runAsNonRoot,
```

Comment:
the `FSGroup` is not set on the `want` object in this test case because we do not allow rootGroup 0 to be set unless the user has set the insecure allow full deployment flag, right?